src/apps/backend/modules/password_reset_token/internal/password_reset_token_reader.py

```diff
@@ -0,0 +1,27 @@
+from bson.objectid import ObjectId
+
+from modules.password_reset_token.errors import PasswordResetTokenNotFoundError
+from modules.password_reset_token.internal.store.password_reset_token_repository import PasswordResetTokenRepository
+from modules.password_reset_token.types import PasswordResetToken
+from modules.password_reset_token.internal.password_reset_token_util import PasswordResetTokenUtil
+
+class PasswordResetTokenReader:
+    @staticmethod
+    def get_password_reset_token_by_account_id(account_id: str) -> PasswordResetToken:
+        cursor = PasswordResetTokenRepository.password_reset_token_db.find(
+            {"account": ObjectId(account_id)}
+        ).sort("expires_at", -1)
+
+        try:
+            token_data = next(cursor)
+        except StopIteration:
+            raise PasswordResetTokenNotFoundError()
+
+        return PasswordResetToken(
+            id=str(token_data.get("_id")),
+            is_expired=PasswordResetTokenUtil.is_token_expired(token_data.get("expires_at")),
```

Comment:
Shouldn't `is_token_expired` just accept the `token_data` directly?
We are basically passing everything directly.

Reply:
the function `is_expired` takes parameter of `expires_at` ,which is of type datetime, of `PasswordResetToken` module and it is being used differently in 2 locations so we cannot directly pass `token_data`